pkg/virtctl/network/dynamicifaces_test.go

```diff
@@ -0,0 +1,167 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package network_test
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	fakek8sclient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/testing"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/virtctl/network"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+)
+
+var _ = Describe("Dynamic Interface Attachment", func() {
+	var (
+		ctrl       *gomock.Controller
+		kubeClient *fakek8sclient.Clientset
+		vm         *kubecli.MockVirtualMachineInterface
+		vmi        *kubecli.MockVirtualMachineInstanceInterface
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		DeferCleanup(ctrl.Finish)
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+
+		kubeClient = fakek8sclient.NewSimpleClientset()
+	})
+
+	const (
+		ifaceName   = "pluggediface1"
+		networkName = "newnet"
+		vmName      = "myvm1"
+	)
+
+	DescribeTable("should fail when required input parameters are missing", func(cmdType string, args ...string) {
+		cmd := clientcmd.NewVirtctlCommand(append([]string{cmdType}, args...)...)
+		err := cmd.Execute()
+		Expect(err).To(HaveOccurred())
+	},
+		Entry("missing the VM name as parameter for the `AddInterface` cmd", network.HotplugCmdName),
+		Entry("missing all required flags for the `AddInterface` cmd", network.HotplugCmdName, vmName),
+		Entry("missing the network name flag for the `AddInterface` cmd", network.HotplugCmdName, vmName, "--iface-name", ifaceName),
+		Entry("missing the interface name flag for the `AddInterface` cmd", network.HotplugCmdName, vmName, "--network-name", networkName),
+	)
+
+	It("fails when the VM name argument is missing but all flags are provided", func() {
+		cmd := clientcmd.NewVirtctlCommand(append([]string{network.HotplugCmdName}, requiredCmdFlags(networkName, ifaceName)...)...)
+		err := cmd.Execute()
+
+		const missingArgError = "argument validation failed"
+		Expect(err).To(MatchError(ContainSubstring(missingArgError)))
+	})
+
+	When("all the required input parameters are provided", func() {
+		BeforeEach(func() {
+			kubeClient.Fake.PrependReactor("*", "*", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				Expect(action).To(BeNil())
+				return true, nil, errors.New("kubeClient command not mocked")
+			})
+		})
+
+		DescribeTable("works", func(cmdType string, additionalFlags ...string) {
```

Comment:
I think the test is pretty straight-forward though ... 

I mostly need to know which object type to mock.

Reply:
You right it is, it just that the DescribeTable forces it to have the "if persist" logic.
It would be easier to understand the test  that way IMO.
Not mandatory, your call.

Reply:
I agree with @ormergi , this is a classic case of forcing two unrelated functions to be merged into one.
I mean, this looks to me like this:
```
func foo(flag bool) {
    if flag {
        fooA()
    } else {
        fooB()
    }
}
```

Yes, you do have some common code (3 lines), but the concept still stands.

Just make two different tests and by that reduce the conditions needed.